plugin/trino-hive/src/test/java/io/trino/plugin/hive/fs/TestTransactionScopeCachingDirectoryLister.java

```diff
@@ -96,7 +98,8 @@ public void testConcurrentDirectoryListing()
                         path1, ImmutableList.of(firstFile, secondFile),
                         path2, ImmutableList.of(thirdFile)));
 
-        TransactionScopeCachingDirectoryLister cachingLister = new TransactionScopeCachingDirectoryLister(countingLister, 2);
+        // set concurrencyLevel to 1 as Guava cache with higher concurrencyLimit is not deterministic
```

Comment:
i didn't see this in other tests that rely on Guava caches, for example not in `TestEvictable*Cache` nor `TestCachingJdbcClient`.
i am not saying this isn't true, but i am not convinced by this explanation just yet.
especially given that this test is **single-threaded**. 

can you shed more light on this?

Reply:
explained offline. 

the code comment needs an improvement, as i don't think we're facing a typical non-determinism here.